File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestMonotonicSubarray = function (nums) {
    if (nums.length == 1) return 1
    let inc = 0
    let dec = 0
    let ans = 0
    // Checking increasing
    let cur = 1
    for (let i = 1; i < nums.length; i++) {
        if (nums[i - 1] < nums[i]) {
            cur++
        }
        else {
            cur = 1
        }
        inc = Math.max(cur, inc)
    }
    // Checking decreasing
    cur = 1
    for (let i = 1; i < nums.length; i++) {
        if (nums[i - 1] > nums[i]) {
            cur += 1
        }
        else {
            cur = 1
        }
        dec = Math.max(cur, dec)
    }
    ans = Math.max(inc, dec)
    return ans
};

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return 1
        
        inc = 0
        dec = 0
        ans = 0
        
        # Checking increasing
        cur = 1
        for i in range(1, len(nums)):
            if nums[i - 1] < nums[i]:
                cur += 1
            else:
                cur = 1
            inc = max(cur, inc)
        
        # Checking decreasing
        cur = 1
        for i in range(1, len(nums)):
            if nums[i - 1] > nums[i]:
                cur += 1
            else:
                cur = 1
            dec = max(cur, dec)
        
        ans = max(inc, dec)
        return ans